sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Shared/ConnectionStatesConverter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+// This is shared between WPS.Common and WPS.AspNetCore
+
+namespace Microsoft.Azure.WebPubSub.Common
+{
+    /// <summary>
+    /// Converter to turn the ConnectionStates dictionary into a regular JSON
+    /// object works for data transport over http between server and service.
+    /// </summary>
+    internal class ConnectionStatesConverter : JsonConverter<IReadOnlyDictionary<string, BinaryData>>
+    {
+        /// <inheritdoc/>
+        public override IReadOnlyDictionary<string, BinaryData> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var dic = new Dictionary<string, BinaryData>();
+            var element = JsonDocument.ParseValue(ref reader).RootElement;
+            foreach (var elementInfo in element.EnumerateObject())
+            {
+                var decoded = elementInfo.Value.GetBytesFromBase64();
+                dic.Add(elementInfo.Name, BinaryData.FromBytes(decoded));
+            }
+            return dic;
+        }
+
+        /// <inheritdoc/>
+        public override void Write(Utf8JsonWriter writer, IReadOnlyDictionary<string, BinaryData> value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+            if (value != null)
+            {
+                foreach (KeyValuePair<string, BinaryData> pair in value)
+                {
+                    writer.WriteBase64String(pair.Key, pair.Value.ToArray());
```

Comment:
What does this mean for JS devs?  Do they have to base64 decode everything and then JSON.parse it?

Reply:
This mainly works between server/service where user probably set state with `BinaryData.FromString("abc")` or `BinaryData.FromBytes()` where would not able to get by `BinaryData.ToObjectFromJson<T>()`, and base64 encode avoids data loss for binary.
In functions, we use `ConnectionStatesNewtonsoftConverter` to populate the value to JS devs which won't do base64 and always ensure `FromObjectAsJson` and `ToObjectFromJson`.